### DISCUSSION TITLE
Database Structure

### DISCUSSION THREAD

**Discussion by JaksyMarci** [JaksyMarci/Cherry-Blossom-Diner#2] [General]
We need to sketch up the structure of the database soon.

**Comment by JaksyMarci** [comment]
My initial idea (copy-pase this to dbdiagram.io)
Still a rough sketch, I'll add primary keys and some other relations later.

Table Diner.tables {
  tableId int
  numberOfSeats int
  state enum
  waiterId int

}

Table Diner.waiters {
  waiterId int
  waiterName varchar2
  waiterPassword varchar2
  
}

Table Diner.invoices {
  invoiceId int
  tableId int
  amount double
  
}

Table Diner.Menu{
  foodName varchar2
  foodPrice double
  
}



Ref: "Diner"."tables"."waiterId" < "Diner"."waiters"."waiterId"

Ref: "Diner"."invoices"."tableId" - "Diner"."tables"."tableId"

**Reply by StavniczkiBalazs** [reply]
I've made a few changes about the structure, because it seemed to be easier to me with these updates.
Our DB structure now:
Table Menus {
id
food_name
food_price
food_type
created_at
updated_at
}

Table Tables {
id
number_of_seats
state
bill
user_id ->foreign key -> User.id
created_at
updated_at
}

Table Users {
id
name
email
email_verified_at
password
is_admin
remember_token
created_at
verified_at
}

Pivot Table Table_food {
id
table_id ->foreign key -> Table.id
food_id ->foreign key -> Food.id
created_at
updated_at
}

**Comment by JaksyMarci** [comment]
My final preposition

![image](https://user-images.githubusercontent.com/70854029/194523438-b45cec6b-9833-4a1e-a849-8e606f6db8b2.png)


Table Diner.tables {
tableId int [pk]
numberOfSeats int
state enum


}

Table Diner.waiter {
waiterId int [pk]
waiterName varchar2
waiterPassword varchar2

}

Table Diner.invoices {
invoiceId int [pk]
tableId int
amount double
isPaid boolean

}

Table Diner.Menu{
foodName varchar2 [pk]
foodPrice double
foodType varchar2

}

Table Diner.idk{
  waiterId int [pk]
  tableId int
  
}


Ref: "Diner"."invoices"."tableId" - "Diner"."tables"."tableId"

Ref: "Diner"."idk"."waiterId" - "Diner"."waiter"."waiterId"

Ref: "Diner"."idk"."tableId" < "Diner"."tables"."tableId"
